cli/pkg/github/auth.go

```diff
@@ -0,0 +1,140 @@
+package github
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+// DeviceFlow allows to authorize users to GitHub.
+// See https://docs.github.com/en/developers/apps/authorizing-oauth-apps#device-flow.
+type DeviceFlow struct {
+	baseURI *url.URL
+	client  *http.Client
+}
+
+// NewDeviceFlow creates a new GitHub device authorization flow.
+func NewDeviceFlow(baseURL *url.URL, client *http.Client) DeviceFlow {
+	return DeviceFlow{baseURI: baseURL, client: client}
+}
+
+// GetCodes requests the device and user verification codes from GitHub.
+func (df DeviceFlow) GetCodes(req *GetCodesRequest) (*GetCodesResponse, error) {
+	uri := df.uri("login/device/code")
+	resp := new(GetCodesResponse)
+	if err := df.do(http.MethodPost, uri, req, resp); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// GetCodesRequest contains input parameters to request the device and user
+// verification codes from GitHub.
+type GetCodesRequest struct {
+	ClientID string `json:"client_id"`
+	Scope    string `json:"scope,omitempty"`
+}
+
+// GetCodesResponse contains response parameters when requesting the device
+// and user verification codes from GitHub.
+type GetCodesResponse struct {
+	DeviceCode      string `json:"device_code"`
+	UserCode        string `json:"user_code"`
+	VerificationURI string `json:"verification_uri"`
+	ExpiresIn       int    `json:"expires_in"`
+	Interval        int    `json:"interval"`
+}
+
+// GetAccessToken requests the access token for the authorized user.
+func (df DeviceFlow) GetAccessToken(
+	req *GetAccessTokenRequest,
+) (*GetAccessTokenResponse, error) {
+	uri := df.uri("login/oauth/access_token")
```

Comment:
Would you mind adding this path to `cli/pkg/github/github.go` and use it as a constant as well? By doing this, if it changes, we'd need to simply change the value in `github.go`. Same as the `baseURL` variable.